Services/DataX.Flow/Flow.ManagementService/Controllers/FlowManagementController.cs

```diff
@@ -463,5 +481,42 @@ public async Task<ApiResult> SyncJobsByNames([FromBody] string[] jobNames)
             }
         }
 
+        [HttpPost]
+        [Route("job/waittilljobsrunning")]
+        public async Task<ApiResult> WaitTillJobsRunning([FromBody] WaitTillJobsRunningRequest waitRequest)
+        {
+            try
+            {
+                RolesCheck.EnsureWriter(Request, _isLocal);
+
+                var maxRetryTimes = waitRequest.MaxRetryTimes;
+                var jobNames = waitRequest.JobNames;
+                var intervalsBetweenRetriesInSeconds = waitRequest.IntervalsBetweenRetriesInSeconds;
+
+                int retryTimes = 0;
+                SparkJobFrontEnd[] jobOpResult = null;
+                while (retryTimes++ < maxRetryTimes)
+                {
+                    jobOpResult = jobNames != null && jobNames.Any() ?
+                                await _jobOperation.SyncJobStateByNames(jobNames)
+                                : await _jobOperation.SyncAllJobState();
+
+                    var startingJobs = jobOpResult.Where(r => r.JobState == JobState.Starting);
+                    if (!startingJobs.Any())
```

Comment:
in case the job is not running, shouldn't we issue a restart again for that job?

Reply:
yes, it would restart when job not running

Reply:
yes, will restart it